packages/ui/tests/component/button/button.cy.ts

```diff
@@ -1,72 +1,149 @@
-    
+
     // Act
-    // @ts-ignore
-    cy.mount(Button, {
-      props: {
-        id: id,
-        label: label
-      }
-    })
-    
+    mountButton({ id: id, label: label })
+
     // Assert
     cy.get(`#${id}`).should('be.visible').and('contain.text', label)
   })
   
-  describe('Base variants', () => {
-    ;[
-      { title: "Applies primary button classes when severity is null", severity: null },
-      { title: "Applies primary button classes when severity is primary", severity: "primary" },
-      { title: "Applies secondary button classes when severity is secondary", severity: "secondary" },
-      { title: "Applies success button classes when severity is success", severity: "success" },
-      { title: "Applies info button classes when severity is info", severity: "info" },
-      { title: "Applies warning button classes when severity is warning", severity: "warning" },
-      { title: "Applies danger button classes when severity is danger", severity: "danger" },
-    ]
-      .forEach(({ title, severity }) => {
-        it(title, () => {
+  describe('Variants', () => {
+    ;[{ severity: undefined, rgb: 'rgb(0, 119, 200)' }, { severity: "primary", rgb: 'rgb(0, 119, 200)' }, 
+      { severity: "secondary", rgb: 'rgb(95, 95, 95)' }, { severity: "success", rgb: 'rgb(33, 170, 100)' },
+      { severity: "info", rgb: 'rgb(13, 202, 240)' }, { severity: "warning", rgb: 'rgb(255, 174, 0)' }, 
+      { severity: "danger", rgb: 'rgb(247, 35, 35)' }
+    ].forEach(({ severity, rgb }) => {
+
+      describe(severity !== undefined ? severity : 'not specified', () => {
+        it(`Applies ${severity !== undefined ? severity : 'primary'} button classes when severity is ${severity}`, () => {
+          // Arrange
+          const id = "button"
+          const label = "Button"
+          const expectedClass = severity !== undefined ? `btn btn-${severity}` : 'btn btn-primary'
+
+          // Act
+          mountButton({ id: id, label: label, severity: severity })
+
+          // Assert
+          cy.get(`#${id}`).should('have.class', expectedClass)
+        })
+      
+        it(`Applies ${severity !== undefined ? severity : 'primary'} outline button classes when severity is ${severity} and outline is true`, () => {
+          // Arrange
+          const id = "button"
+          const label = "Button"
+          const expectedClass = severity !== undefined ? `btn btn-outline-${severity}` : 'btn btn-outline-primary'
+
+          // Act
+          mountButton({ id: id, label: label, severity: severity, outlined: true })
+
+          // Assert
+          cy.get(`#${id}`).should('have.class', expectedClass)
+        })
+        
+        it(`Displays a ${severity !== undefined ? severity : 'primary'} badge when severity is ${severity}`, () => {
+          // Arrange
+          const id = "button"
+          const label = "Button"
+          const expectedClass = 'badge'
+          const expectedBackgroundColour = "rgb(255, 255, 255)"
```

Comment:
do we care what the background colour is?

Reply:
I created separate css classes for the badge to make sure that it would be visible depending on the components severity I.e. primary will add a white background to the badge whereas the primary outline variant will add a blue background. So I wanted to make sure that the background is white as if for example it is on a primary button the badge won’t be visible as the text will be blue.